lib/Driver.js

```diff
@@ -46,34 +53,28 @@ module.exports = class Driver {
     }
 
     const [didAuthority, keyIdFragment] = did.split('#');
-
-    let addedToCache = false;
-    let promise = this._cache.get(didAuthority);
-    if(!promise) {
-      const fingerprint = didAuthority.substr('did:key:'.length);
-      const publicKey = LDKeyPair.fromFingerprint({fingerprint});
-
-      promise = this.keyToDidDoc(publicKey);
-      this._cache.set(didAuthority, promise);
-      addedToCache = true;
-    }
-
-    let didDoc;
     try {
-      didDoc = await promise;
-    } catch(e) {
-      if(addedToCache) {
-        this._cache.del(didAuthority);
+      // cache the result of expansion
+      const {didDocument} = await this._cache.memoize({
+        key: didAuthority,
+        fn: () => {
+          const fingerprint = didAuthority.substr('did:key:'.length);
+          const keyPair =
+            Ed25519VerificationKey2018.fromFingerprint({fingerprint});
+          return this.keyPairToDidDocument({keyPair});
```

Comment:
I don't suggest we do anything here right now, but I do want to point out how strange this looks. You can't get a key pair from a public key fingerprint. If you could, it would be a serious security issue.